fastn-update/src/lib.rs

```diff
@@ -2,142 +2,82 @@ use std::io::Read;
 
 extern crate self as fastn_update;
 
-static GITHUB_PAGES_REGEX: once_cell::sync::Lazy<regex::Regex> =
-    once_cell::sync::Lazy::new(|| regex::Regex::new(r"([^/]+)\.github\.io/([^/]+)").unwrap());
-
-fn extract_github_details(pages_url: &str) -> Option<(String, String)> {
-    if let Some(captures) = GITHUB_PAGES_REGEX.captures(pages_url) {
-        let username = captures.get(1).unwrap().as_str().to_string();
-        let repository = captures.get(2).unwrap().as_str().to_string();
-        Some((username, repository))
-    } else {
-        None
-    }
-}
-
-// https://api.github.com/repos/User/repo/:archive_format/:ref
-// https://stackoverflow.com/questions/8377081/github-api-download-zip-or-tarball-link
-async fn resolve_dependency_from_gh(
-    username: &str,
-    repository: &str,
-) -> fastn_core::Result<Vec<u8>> {
-    let url = format!(
-        "https://api.github.com/repos/{}/{}/zipball",
-        username, repository
-    );
-    let zipball = fastn_core::http::http_get(&url).await?;
-    Ok(zipball)
-}
-
-#[derive(serde::Serialize, serde::Deserialize)]
-pub struct Manifest {
-    pub packages: Vec<Package>,
-}
-
-impl Manifest {
-    pub fn new(packages: Vec<Package>) -> Self {
-        Manifest { packages }
-    }
-}
-
-#[derive(serde::Serialize, serde::Deserialize)]
-pub enum PackageSource {
-    Github {
-        username: String,
-        repository: String,
-    },
-}
+mod types;
+mod utils;
+
+pub async fn resolve_dependencies(
+    config: &fastn_core::Config,
+    packages_root: &camino::Utf8PathBuf,
+    dependencies: &Vec<fastn_core::package::dependency::Dependency>,
+    packages: &mut Vec<fastn_update::types::Package>,
+) -> fastn_core::Result<()> {
+    for dependency in dependencies {
+        match fastn_update::utils::get_package_source_url(&dependency.package) {
+            Some(source) => {
+                let (mut archive, checksum) =
+                    fastn_update::utils::download_archive_from_gh(source.clone()).await?;
```

Comment:
This should not be `github` specific. It must work as long as you have a zip file URL available. The fact that we are using Github or Github is generating should not be your concern/assumption.

Reply:
yes, I have refactored this in the latest commit